product/02-innersource-and-agile.md

```diff
@@ -0,0 +1,125 @@
+
+Well facored and well tested code is one goal of any agile team. In an InnerSource setting not only does that lead to faster on-ramp times for team members, it helps getting contributors up to speed quickly as well.
+
+Teams that are already familiar with collaborative ways of working together and avoid assigning individual tasks are in a good position to also deal with external contributions in a flexible manner.
+They also bring a mindset and communication style that works well for motivating contributors over whose priority they have no direct influence.
+Working with intrinsic motivation instead of directing work means that host teams already have the tools that will help the successfully work with contributors.
+
+If teams are already successfully pairing to work on problems they are already comfortable with sharing progress early for the purpose.
+One challenge moving to InnerSource from a pairing only culture often is two-fold: The host team needs to make time for supporting contributors and schedule that into their planned work.
+In addition when crossing team boundaries it is often hard to find time slots for pairing - in those cases it should be complemented with asynchronous collaboration.
+To avoid frequent disruption time for that should also be set aside.
+Often it is simplest to set aside certain hours in the day or a day a week for that work.
+Making that explicit at the team level takes a lot of pressure off of engineers trying to fullfill their own team goals but also helping out contributors.
+Another challenge with pairing is that it allows pairs to move very quickly together - often at the expense of writing important information down for the rest of the team.
+In an InnerSource setting it does take training to remember to bring all relevant decisions back to shared communication channels that both, host team and contributors can follow.
+From a product perspective that does bring a lot more transparency to the development process.
+It also means that decisions that otherwise may have been taken at the engineering level only are now visible for everyone involved.
+
+Remember last time you insisted that your product be well tested, preferably with automated tests so deployments can happen frequently and without human intervention?
+This goal now helps with InnerSource as well: Contributions are much easier if contributors can check locally if their changes are safe.
+Tests also ensure that the host team remembers to keep the contributed functionality if they are reminded of the reason for it by a failing test.
+
+Remember last time you insisted on your team spending time to follow the goal of "leave code in a better shape than you found it"?
+That mindset will also help you when moving to an InnerSource model: It makes sure that quality and co-hesion of code remains high even when there are multiple contributions from different sources.
+
+
+## Common misunderstandings when coming from agile teams
```

Comment:
### Additional common misunderstandings

* "My engineers will now work on whatever projects they like, so I cannot benefit from their work any more / plan with their capacity"
* "I do not need to take great care in protecting my infrastructure credentials used in my projects as we are all a big InnerSource family"
* "I do not have to adopt my conversation tone and can trust that anybody who has access to my project will always keep company secrets"

Reply:
Those are really valuable to have. Thank you for sharing those!